code/ship/ship.cpp

```diff
@@ -11234,6 +11233,16 @@ int ship_fire_primary(object * obj, int stream_weapons, int force)
 							// of weapon_create							
 							weapon_objnum = weapon_create( &firing_pos, &firing_orient, weapon_idx, OBJ_INDEX(obj), new_group_id,
 								0, 0, swp->primary_bank_fof_cooldown[bank_to_fire] );
+
+							if (weapon_objnum == -1) {
+								// Weapon most likely failed to fire
+								if (obj == Player_obj) {
+									ship_maybe_play_primary_fail_sound();
+								}
+								//has_fired = true; // Just because it failed, doesn't mean it wouldn't have fired
```

Comment:
Why is this line commented out?

Reply:
I'm still waiting on Axem to get back to me on whether a weapon failure should trigger the `On_Weapon_Fired` scripting hook.

Reply:
Ok. In that case you should probably add the "unfinished" label so that it is visible that these changes are not final yet.